app/javascript/mastodon/components/media_gallery.jsx

```diff
@@ -92,14 +100,28 @@ class Item extends PureComponent {
     let badges = [], thumbnail;
 
     let width  = 50;
-    let height = 100;
+    let height = 50;
 
-    if (size === 1) {
-      width = 100;
-    }
+    const cols = colCount(size);
+    const remaining = (-size % cols + cols) % cols;
 
-    if (size === 4 || (size === 3 && index > 0)) {
-      height = 50;
+    if (remaining === 3 && index === 0 || size === 1) {
+      width = 100;
+      height = 100;
+    } else if (remaining === 2 && index < 2) {
+      if (cols >= 4) {
+        width = 100;
```

Comment:
Maybe, always using tall item could be better then wide image.
It is already 3/2 wide image. 6/2 is weired while 3/4 looks good

Also on remaining == 1

Reply:
Additionally, it could be more simplified if using tall items always